management/authenticators.go

```diff
@@ -0,0 +1,329 @@
+package ziti
+
+import (
+	"context"
+	"crypto"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"github.com/go-openapi/runtime"
+	"github.com/go-openapi/strfmt"
+	"github.com/openziti/edge-api/rest_management_api_client/authentication"
+	"github.com/openziti/edge-api/rest_model"
+	"github.com/openziti/identity"
+	"github.com/openziti/sdk-golang/ziti"
+	"github.com/openziti/sdk-golang/ziti/sdkinfo"
+	"net/url"
+)
+
+type ApiAuthenticator interface {
```

Comment:
It is similar but the root issue is to go-swagger generated code. It isn't implemented in a way where I can reference both clients via an interface without defining an interface through wrapper structs. It is possible, it is just kinda ugly and that interface is largely useless because the generated clients have vastly different capabilities.

So there is a trade-off, dupe code with similar naming or wrapper structs with interfaces and generics.